docs/_basic/listening_modals.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Listening for view submissions
+lang: en
+slug: view_submissions
+order: 12
+---
+
+<div class="section-content">
+
+If a <a href="https://api.slack.com/reference/block-kit/views">view payload</a> contains any input blocks, you must listen to `view_submission` events to receive their values. To listen to `view_submission` events, you can use the built-in `view()` method. `view()` requires a `callback_id` of type `str` or `RegExp`.
+
+You can access the value of the `input` blocks by accessing the `state` object. `state` contains a `values` object that uses the `block_id` and unique `action_id` to store the input values.
+
+Read more about view submissions in our <a href="https://api.slack.com/surfaces/modals/using#interactions">API documentation</a>.
+
+</div>
+
+```python
+# Handle a view_submission event
+@app.view("view_b")
+def handle_submission(ack, body, client, view):
+    # Acknowledge the view_submission event
+    ack()
+
+    # Do whatever you want with the input data - here we're saving it to a DB then sending the user a verifcation of their submission
+
+    # Assume there's an input block with `block_1` as the block_id and `input_a`
+    val = view["state"]["values"]["block_1"]["input_a"]
+    user = body["user"]["id"]
+
```

Comment:
hmm, why don't we have an input validation sample here? We should do this (also for Bolt JS)

Reply:
Not sure! I think it's a good idea, but again, just ported over BoltJS in the interest of time.

Reply:
Thoughts on this? I pulled the validation example we use for `Acknowledging Events`, so if we want a different one, I'm open to alternatives/suggestions. cc: @shaydewael 

```

    # Assume there's an input block with `block_1` as the block_id and `email_input`
    email = view['state']['values']['block_1']['email_input']
    user = body['user']['id']

    is_email = "^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$"
    if re.match(is_email, email):
        # Acknowledge the view_submission event
        ack()
    else:
        # If it isn’t a valid email, acknowledge with an error
        errors = [{
            "name": "email_address",
            "error": "Sorry, this isn’t a valid email"
        }]
        ack(errors=errors)
```

Reply:
I also have never done this type of validation with regard to `view_submission` first-hand yet, so I'm not even sure the above code is applicable to this case. Please advise!

Reply:
@misscoded How about having simpler ones such as length-check? Here is an example I recently created: https://github.com/seratch/apidays-workshop-2020/blob/87f623356f90fbdb9f5112e114dc51c157698ac7/python/app.py#L272-L283